src/Microsoft.OData.ModelBuilder/Vocabularies/VocabularyTermConfigurationExtensions.cs

```diff
@@ -233,6 +261,509 @@ public static ModificationQueryOptionsConfiguration HasModificationQueryOptions(
+        /// <returns><see cref="OrderedConfiguration"/></returns>
+        public static OrderedConfiguration HasOrdered<TEntity>(this EntitySetConfiguration<TEntity> navigationSource) where TEntity : class
+            => navigationSource?.Configuration.VocabularyTermConfigurations.GetOrCreateConfiguration<OrderedConfiguration>();
+
+        /// <summary>
+        /// <see cref="OrderedConfiguration"/> configuration
+        /// </summary>
+        /// <param name="vocabularyConfigurationsBase">The <see cref="IEdmVocabularyAnnotatable"/> that can be built using <see cref="ODataModelBuilder"/>.</param>
+        /// <returns><see cref="OrderedConfiguration"/></returns>
+        public static OrderedConfiguration HasOrdered(this VocabularyConfigurationsBase vocabularyConfigurationsBase)
+            => vocabularyConfigurationsBase?.VocabularyTermConfigurations.GetOrCreateConfiguration<OrderedConfiguration>();
+
+        /// <summary>
+        /// <see cref="PositionalInsertConfiguration"/> configuration
+        /// </summary>
+        /// <typeparam name="TEntity">The entity type of the navigation source.</typeparam>
+        /// <param name="navigationSource">The <see cref="IEdmNavigationSource"/> that can be built using <see cref="ODataModelBuilder"/>.</param>
+        /// <returns><see cref="PositionalInsertConfiguration"/></returns>
+        public static PositionalInsertConfiguration HasPositionalInsert<TEntity>(this EntitySetConfiguration<TEntity> navigationSource) where TEntity : class
+            => navigationSource?.Configuration.VocabularyTermConfigurations.GetOrCreateConfiguration<PositionalInsertConfiguration>();
+
+        /// <summary>
+        /// <see cref="PositionalInsertConfiguration"/> configuration
+        /// </summary>
+        /// <param name="vocabularyConfigurationsBase">The <see cref="IEdmVocabularyAnnotatable"/> that can be built using <see cref="ODataModelBuilder"/>.</param>
+        /// <returns><see cref="PositionalInsertConfiguration"/></returns>
+        public static PositionalInsertConfiguration HasPositionalInsert(this VocabularyConfigurationsBase vocabularyConfigurationsBase)
+            => vocabularyConfigurationsBase?.VocabularyTermConfigurations.GetOrCreateConfiguration<PositionalInsertConfiguration>();
+
+        /// <summary>
+        /// <see cref="AlternateKeysConfiguration"/> configuration
+        /// </summary>
+        /// <typeparam name="TEntity">The entity type of the navigation source.</typeparam>
+        /// <param name="navigationSource">The <see cref="IEdmNavigationSource"/> that can be built using <see cref="ODataModelBuilder"/>.</param>
+        /// <returns><see cref="AlternateKeysConfiguration"/></returns>
+        public static AlternateKeysConfiguration HasAlternateKeys<TEntity>(this EntitySetConfiguration<TEntity> navigationSource) where TEntity : class
```

Comment:
>EntitySetConfiguration [](http://example.com/codeflow?start=80&length=22)

why some extension method are built on EntitySetConfiguration<T>, others are built on VocabularyConfigurationsBase?

Reply:
@xuzhg the extension methods that  are built on VocabularyConfigurationBase are those configurations that are not of type<T>. The EntitySetConfiguration is of typeT like EntitySetConfiguration<TEntity> but PropertyConfiguration, EnumMemberConfiguration, etc are not of typeT